x-pack/test/functional/apps/dashboard/drilldowns/dashboard_drilldowns.ts

```diff
@@ -0,0 +1,166 @@
+      await dashboardDrilldownsManage.expectsCreateDrilldownFlyoutOpen();
+      await dashboardDrilldownsManage.fillInDashboardToDashboardDrilldownWizard({
+        drilldownName: DRILLDOWN_TO_AREA_CHART_NAME,
+        destinationDashboardTitle: DASHBOARD_WITH_AREA_CHART_NAME,
+      });
+      await dashboardDrilldownsManage.saveChanges();
+      await dashboardDrilldownsManage.expectsCreateDrilldownFlyoutClose();
+
+      // check that drilldown notification badge is shown
+      expect(await PageObjects.dashboard.getPanelDrilldownCount()).to.be(1);
+
+      // save dashboard, navigate to view mode
+      await PageObjects.dashboard.saveDashboard(DASHBOARD_WITH_PIE_CHART_NAME, {
+        saveAsNew: false,
+        waitDialogIsClosed: true,
+      });
+
+      // trigger drilldown action by clicking on a pie and picking drilldown action by it's name
+      await pieChart.filterOnPieSlice('40,000');
+      await dashboardDrilldownPanelActions.expectMultipleActionsMenuOpened();
+      await navigateWithinDashboard(async () => {
+        await dashboardDrilldownPanelActions.clickActionByText(DRILLDOWN_TO_AREA_CHART_NAME);
+      });
+
+      // check that we drilled-down with filter from pie chart
+      expect(await filterBar.getFilterCount()).to.be(1);
+
+      const originalTimeRangeDurationHours = await PageObjects.timePicker.getTimeDurationInHours();
+
+      // brush area chart and drilldown back to pie chat dashboard
+      await brushAreaChart();
+      await dashboardDrilldownPanelActions.expectMultipleActionsMenuOpened();
+      await navigateWithinDashboard(async () => {
+        await dashboardDrilldownPanelActions.clickActionByText(DRILLDOWN_TO_PIE_CHART_NAME);
+      });
+
+      // because filters are preserved during navigation, we expect that only one slice is displayed (filter is still applied)
+      expect(await filterBar.getFilterCount()).to.be(1);
+      await pieChart.expectPieSliceCount(1);
+
+      // check that new time range duration was applied
+      const newTimeRangeDurationHours = await PageObjects.timePicker.getTimeDurationInHours();
+      expect(newTimeRangeDurationHours).to.be.lessThan(originalTimeRangeDurationHours);
+
+      // delete drilldown
+      await PageObjects.dashboard.switchToEditMode();
+      await dashboardPanelActions.openContextMenu();
+      await dashboardDrilldownPanelActions.expectExistsManageDrilldownsAction();
+      await dashboardDrilldownPanelActions.clickManageDrilldowns();
+      await dashboardDrilldownsManage.expectsManageDrilldownsFlyoutOpen();
+
+      await dashboardDrilldownsManage.deleteDrilldownsByTitles([DRILLDOWN_TO_AREA_CHART_NAME]);
+      await dashboardDrilldownsManage.closeFlyout();
+
+      // check that drilldown notification badge is shown
+      expect(await PageObjects.dashboard.getPanelDrilldownCount()).to.be(0);
+    });
+
+    it('browser back/forward navigation works after drilldown navigation', async () => {
```

Comment:
This covers simple case with simple dashboard. 
There is an edge case: if embeddable update input in the beginning it causes state update and causes redundant history record. This happens, for example, if embeddable decides to do initial state migrations from older versions. 
Need to follow up separately on those edge cases with back button in dashboard